src/client/datascience/interactive-common/notebookProvider.ts

```diff
@@ -176,7 +176,8 @@ export class NotebookProvider implements INotebookProvider {
     }
 
     private fireConnectionMade() {
-        this._connectionMade.fire();
+        // Disable the UI to avoid errors before the user runs a cell
```

Comment:
I don't think this  should always be true. It should be based on the options passed to the connect call.

Reply:
In the spots where this is bound, you can likely bind it to the options passed into the caller.

Reply:
Sorry it looks like there are two more spots. In src\client\datascience\interactive-common\notebookProvider.ts where the connectionMade event is bound to the fireConnectionMade function. You should bind the first parameter to the options.DisableUI setting.

Reply:
Line 86 and 95 in notebookProvider.ts